### ISSUE TITLE
New built-in function: joining a list of strings

### ISSUE THREAD

**Issue by saig0** [camunda/feel-scala#327]
**Is your feature request related to a problem? Please describe.**
I have a list of strings and want to join/combine the strings into a single string.

Similar to Java's [joining](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/stream/Collectors.html#joining(java.lang.CharSequence,java.lang.CharSequence,java.lang.CharSequence)) function:

```
joining(["a","b","c"])
// "abc"

joining(["a","b","c"], ", ")
// "a, b, c"

joining(["a","b","c"], ", ", "[", "]")
// "[a, b, c]"
```

Currently, there is no way to join the strings, or it is very complicated.

**Describe the solution you'd like**

A new built-in function with the following signature: 

```
joining(list: list<string>, delimiter: string, prefix: string, suffix: string): string
```

* `list` - the list of strings to join
* `delimiter`- (optional) the string that is used between each element (default: `""` empty string)
* `prefix`- (optional) the string that is used at the beginning of the joined result (default: `""` empty string)
* `suffix`- (optional) the string that is used at the end of the joined result (default: `""` empty string)

**Related issues**

the idea came up in the context of #315

* Camunda BPM: 
* Zeebe broker:

**Comment by P3trur0** [comment]
May I take this as well?

**Comment by P3trur0** [comment]
Hi @saig0, here when you refer to optional fields are you considering the following behavior through overloaded invocations?

`joining(list: list<string>, delimiter: string, prefix: string, suffix: string)`
`joining(list: list<string>, delimiter: string, prefix: string)`
`joining(list: list<string>, delimiter: string)`
`joining(list: list<string>)`

or do you expect any possible combination of the optional fields?

e.g., is the following admittable?

`joining(list: list<string>, prefix: string, suffix: string)`

As you may notice here we're not passing the second parameter...

?

**Comment by saig0** [comment]
@P3trur0 good point!

> or do you expect any possible combination of the optional fields?

We can limit the method to the three most common use cases:

* `joining(list: list<string>)` - only the list
* `joining(list: list<string>, delimiter: string)` - no prefix and suffix
* `joining(list: list<string>, delimiter: string, prefix: string, suffix: string)` - all arguments

**Comment by P3trur0** [comment]
actually I've implemented all the possible combinations 😄 in #339
